### PR TITLE
improve test-yml builder

### PR DESCRIPTION
**Pull request by KevinMenden**
Running tests twice for `nf-core modules create-test-yml` to catch unreproducible md5 sums.

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/modules/test_yml_builder.py [hunk 1]
```diff
@@ -258,11 +274,15 @@ def run_tests_workflow(self, command):
...
 
         tmp_dir = tempfile.mkdtemp()
+        tmp_dir_repeat = tempfile.mkdtemp()
         command += f" --outdir {tmp_dir}"
+        command_repeat = command + f" --outdir {tmp_dir_repeat}"
 
         log.info(f"Running '{self.module_name}' test with command:\n[violet]{command}")
         try:
             nfconfig_raw = subprocess.check_output(shlex.split(command))
+            nfconfig_raw = subprocess.check_output(shlex.split(command_repeat))
```
Comment (ewels) [inline nf_core/modules/test_yml_builder.py:284]: More logs please, as these can be quite slow 👍🏻

File: nf_core/modules/test_yml_builder.py [hunk 2]
```diff
@@ -227,6 +227,22 @@ def get_md5_sums(self, entry_point, command):
...
+                elem_md5 = self._md5(elem)
+                elem = elem.replace(results_dir_repeat, "output")
+                test_files_repeat.append({"path": elem, "md5sum": elem_md5})
+
+        # Compare both test.yml files
+        test_files.sort()
+        test_files_repeat.sort()
+        for i in range(len(test_files)):
+            if not test_files[i]["md5sum"] == test_files_repeat[i]["md5sum"]:
+                test_files[i].pop("md5sum")
```
Comment (ewels) [inline nf_core/modules/test_yml_builder.py:244]: log please

**Review by ewels — APPROVED** [review]
The OCD part of me kind of want this to be done as a function that is run twice instead of duplicating the lines of code within the function. But it's a very minor thing and now I've said this it's off my chest so happy to approve as-is 😆

File: nf_core/modules/test_yml_builder.py (hunk 1 above)
Comment (KevinMenden) [reply nf_core/modules/test_yml_builder.py:284]: yes good point

File: nf_core/modules/test_yml_builder.py (hunk 2 above)
Comment (KevinMenden) [reply nf_core/modules/test_yml_builder.py:244]: :+1:

**Comment by KevinMenden** [comment]
No you're right it shouldn't be too much of an issue to pack that into a function.
I was thinking about doing that when wondering if we should make this run in separate threads/processes, to speed it up :thinking: 

Might be something for the future, but we'd need to pack it into a function anyway.

**Pull request merged**